functions/imagemagick/ImageMagick/Function.cs

```diff
@@ -0,0 +1,171 @@
+
+            if (annotation.Adult == Likelihood.VeryLikely || annotation.Violence == Likelihood.VeryLikely)
+            {
+                _logger.LogInformation("Detected {uri} as inappropriate.", gcsUri);
+                await BlurImageAsync(data, cancellationToken);
+            }
+            else
+            {
+                _logger.LogInformation("Detected {uri} as OK.", gcsUri);
+            }
+        }
+        // [END functions_imagemagick_analyze]
+
+        // [START functions_imagemagick_blur]
+        /// <summary>
+        /// Downloads the Storage object specified by <paramref name="data"/>,
+        /// blurs it using ImageMagick, and uploads it to the "blurred" bucket.
+        /// </summary>
+        private async Task BlurImageAsync(StorageObjectData data, CancellationToken cancellationToken)
+        {
+            // Download image
+            string originalImageFile = Path.GetTempFileName();
+            using (Stream output = File.Create(originalImageFile))
+            {
+                await _storageClient.DownloadObjectAsync(data.Bucket, data.Name, output, cancellationToken: cancellationToken);
+            }
+
+            // Construct the ImageMagick command
+            string blurredImageFile = Path.GetTempFileName();
+            // Command-line arguments for ImageMagick.
+            // Paths are wrapped in quotes in case they contain spaces.
+            string arguments = $"\"{originalImageFile}\" -blur 0x8, \"{blurredImageFile}\"";
+            ProcessStartInfo psi = new ProcessStartInfo("convert", arguments);
+
+            Process process = Process.Start("convert", arguments);
+            // Process doesn't expose a way of asynchronously waiting for completion.
+            // See https://stackoverflow.com/questions/470256 for examples of how
+            // this can be achieved using events, but for the sake of brevity,
+            // this sample just waits synchronously.
+            process.WaitForExit();
+
+            // If ImageMagick failed, log the error but complete normally to avoid retrying.
+            if (process.ExitCode != 0)
+            {
+                _logger.LogError("ImageMagick exited with code {exitCode}", process.ExitCode);
+                return;
+            }
+
+            // Upload image to blurred bucket.
+            using (Stream input = File.OpenRead(blurredImageFile))
+            {
+                await _storageClient.UploadObjectAsync(
+                    s_blurredBucketName, data.Name, data.ContentType, input, cancellationToken: cancellationToken);
+            }
+
+            string uri = $"gs://{s_blurredBucketName}/{data.Name}";
+            _logger.LogInformation("Blurred image uploaded to: {uri}", uri);
+
+            // Remove images from the file system.
+            File.Delete(originalImageFile);
+            File.Delete(blurredImageFile);
+        }
+        // [END functions_imagemagick_blur]
+        // [START functions_imagemagick_setup]
```

Comment:
I'm fine with leaving this as is. But one alternative could be to have a function called AnalyzeImageAsync and from HandleAsync only call that function, so on `functions_imagemagick_setup` you can include the whole function code. Then move BlurImageAsync and the new AnalyzeImageAsync to a helper class as static methods and show those in their respectives sections.

Reply:
I think I'd prefer to keep the separation the way it is, as that's what other languages do - but we *could* do all of that with partial classes. Admittedly the partial-ness would show up in the docs too.

Reply:
Yep, partial class is good too. But It's fine to keep as is, it's just the brackets.